pkg/rpcclient/unwrap/unwrap.go

```diff
@@ -155,6 +155,9 @@ func SessionIterator(r *result.Invoke, err error) (uuid.UUID, result.Iterator, e
 	if !ok {
 		return uuid.UUID{}, result.Iterator{}, errors.New("the item is InteropInterface, but not an Iterator")
 	}
+	if (r.Session == uuid.UUID{}) && iter.ID != nil {
+		return uuid.UUID{}, result.Iterator{}, errors.New("server returned iterator ID, but no session ID")
+	}
```

Comment:
> we also have custom iterator unmarshaller that sets iterator ID only if session is enabled on the server

I don't see it depending on session IDs, I think it will unmarshal iterator ID without any issues (as it's there in the reply), but there won't be a session ID. The behavior wrt NeoGo server has not changed, but iterator returned from C# nodes with sessions disabled is useless, here we're just failing fast instead of waiting for the user to try traversing this "iterator" and failing and that step.

Reply:
My bad, that's true.